File: AngbandOS.Interface/PageOfGameMessages.cs
namespace AngbandOS.Core.Interface;

public class PageOfGameMessages
{
    public IGameMessage[] GameMessages { get; }
    public int FirstIndex { get; }
    public int LastIndex { get; }
    public PageOfGameMessages(IGameMessage[] messages, int firstMessageIndex, int lastMessageIndex)
    {
        GameMessages = messages;
        FirstIndex = firstMessageIndex;
        LastIndex = lastMessageIndex;
    }
}

File: AngbandOS.Web/ClientApp/src/app/messages-window/page-of-game-messages.ts
import { GameMessage } from "./game-message";

/// <summary>
/// Represents valid game-messages retrieved from an active game.
/// </summary>
export class PageOfGameMessages {
  /** Returns the index of the most recent message in the page.  This is gameMessages[gameMessages.length - 1]. */
  public firstIndex: number;

  /** Returns the index of the oldest message in the page.  This is gameMessages[0]. */
  public lastIndex: number;

  /** Returns the page of game messages. */
  public gameMessages: string[];

  constructor(firstIndex: number, lastIndex: number, gameMessages: string[]) {
    this.firstIndex = firstIndex;
    this.lastIndex = lastIndex;
    this.gameMessages = gameMessages;
  }
}
